improver_tests/utilities/spatial/test_spatial.py

```diff
@@ -773,7 +794,388 @@ def test_maximum_within_vicinity(grid, radius, landmask, expected_result):
+        # values are returned at all points. The land-sea mask does not affect
+        # the retention of the original mask.
+        (
+            np.ma.masked_array(
+                [[1, 0.0, 1], [1, 1, 1], [1, 1, 1]],
+                mask=[[0, 1, 0], [0, 0, 0], [0, 0, 0]],
+            ),
+            1,
+            np.array([[0, 0, 0], [1, 1, 1], [1, 1, 1]]),
+            np.ma.masked_array(
+                [[1, 1, 1], [1, 1, 1], [1, 1, 1]],
+                mask=[[0, 1, 0], [0, 0, 0], [0, 0, 0]],
+            ),
+        ),
+    ],
+)
+def test_minimum_within_vicinity(grid, radius, landmask, expected_result):
+    """Test that minimum_within_vicinity function returns the expected
+    values and masking for various inputs. Variations tried here are:
+
+      - vicinity processing of a simple array that affects the whole array
+      - vicinity processing of a simple array that affects the corners of the
+        array
+      - vicinity processing of a masked array that returns an identically
+        masked array with no values changed.
+      - vicinity processing with a landmask that prevents any spread of
+        values to the rest of the domain.
+      - vicinity processing with a landmask that isolates the central
+        grid point from the effects of value spreading.
+      - vicinity processing with a complex landmask that leads to a dappled
+        value spread.
+      - vicinity processing with a masked array and a landmask that
+        demonstrates that the original masking is retained and not
+        modified by the landmask.
+    """
+    reference = grid.copy()
+
+    result = minimum_within_vicinity(grid, radius, landmask)
+
+    assert result.data.shape == expected_result.shape
+    assert np.allclose(result, expected_result)
+    assert isinstance(result, type(expected_result))
+    assert result.dtype == np.float64
+    if np.ma.is_masked(reference):
+        assert_array_equal(reference.mask, result.mask)
+
+
+@pytest.mark.parametrize(
+    "grid,radius,landmask,expected_result",
+    [
+        # Vicinity processing, with one non-zero central value resulting
+        # in the whole domain returning values of 1/9
+        (
+            np.array([[0, 0, 0], [0, 1.0, 0], [0, 0, 0]]),
+            1,
+            None,
+            1 / 9 * np.ones((3, 3)),
+        ),
+        # Vicinity processing, with one non-zero central value and a nan in
+        # the corner. The resultant grid evaluates the mean, neglecting any
+        # nan values within the vicinity.
+        (
+            np.array([[np.nan, 0, 0], [0, 1.0, 0], [0, 0, 0]]),
+            1,
+            None,
+            np.array(
+                [[1 / 5, 1 / 7, 1 / 9], [1 / 7, 1 / 8, 1 / 9], [1 / 9, 1 / 9, 1 / 9]]
+            ),
+        ),
+        # Vicinity processing, with one non-zero corner value resulting
+        # in neighbouring cells values of 1 within the limit of the
+        # defined vicinity radius
```

Comment:
I don't quite understand this description though the test seems fine.

Reply:
The dangers of copy-paste. Thanks for being so thorough with this!

Reply:
Haha! I can see what you mean. I can assure you that the tone here was sincere. The state of this PR is significantly better than before your review, so thank you.